client/src/main/java/io/pravega/client/segment/impl/EventSegmentReader.java

```diff
@@ -76,7 +77,7 @@ public default ByteBuffer read() throws EndOfSegmentException, SegmentTruncatedE
      * 
      * @return A future that completes when the request to fill the buffer has returned.
      */
-    public abstract CompletableFuture<Void> fillBuffer();
+    public abstract CompletableFuture<SegmentRead> fillBuffer();
```

Comment:
We don't actually have to make this a SegmentRead. There is no reason that the caller actually needs to see the type. So we can define the interface as `CompletableFuture<Object>` then the implementation can return a SegmentRead but the caller can't abuse the segmentRead to do anything.

Reply:
ok, changed it to ` CompletableFuture<?> fillBuffer();